### DISCUSSION TITLE
Radio Stations: Aggregating Live Items by Publisher

### DISCUSSION THREAD

**Discussion by nathangathright** [Podcastindex-org/podcast-namespace#643] [Ideas]
I already shared this suggestion with TrueFans, but I just wanted to document it here for any other interested parties. If a radio station set up podcast feeds with live items for all their shows and connected them with a publisher feed, then podcast apps could aggregate those live streams together into a publisher-level radio station without any new tags.

Listeners could tune into a publisher’s live stream, stream sats from the current show’s value block, and then switch over to the next show’s live stream + value block. The designated end time for the live stream probably shouldn’t be trusted, but the app could trigger the switch when it receives the starting podping for the next show.

Additionally, apps could display a schedule of upcoming programming across a publisher’s shows based on the child feeds’ pending live items.

**Comment by yoshimo** [comment]
Isn't it more likely that you have one live stream for each station across multiple shows and the news segments that get cut into individual segments for downloading later? I don't see multiple streams for multiple shows happening. The complexity for public radio stations at least would be too overwhelming.
Especially because they want to provide an easy way to jump back to an early part of the programme.

**Reply by samsethi** [reply]
I ran River Radio a public UK radio station that broadcast on DAB.  We used an online service from [Radio King](radioking.com).  (This used to be owned by Podcast Host Aushu who recently sold it???)  We live streamed via Shoutcast to Web, Mobile, Alexa, CarPlay etc. 

So yes we had one continuous stream but we had a schedule for each show which was 1 hour with news and host read ads.  We partnered with Whooshkaa (who were bought by Spotify) who then took our live stream from our schedule and turned them into podcasts and published them to [Spotify](https://open.spotify.com/show/1Qb2snAFcBZ650XET8P7CP?si=a63aa92a09284510 ) and Apple. 

### River Radio Podcast on Spotify

<img width="778" alt="J_R__Richards_-_A_Beautiful_End" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/4117f4ac-7499-4444-9952-3c7265680b0c">

However, when I started River Radio there was no [Live Item Tag](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#live-item).   Had there been, I would have added Live Podcasting to the list of other live broadcast options. 

What I am doing is now is working with [PodToo](https://podtoo.com/) and [Afripods](https://afripods.africa/station/hope-fm) who currently produce podcasts for local radio stations.  They will need to show/explain how these radio stations could also live broadcast their radio shows to podcast apps. And just like Podcasts that use LIT and then publish the recorded show after this is the same for radio stations.  

In the example below we have taken the podcasts from [DRN1](https://www.drn1.com.au/) produced by PodToo and created a **Radio [Publisher](https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#publisher)** page on [TrueFans](https://truefans.fm/publishers/drn1?tab=slate).  

Fans can follow this page and if new podcasts/shows are added they will get a notification.  Equally we turn all the shows into one playlist.  Fans can then save the playlist to their profile and edit the playlist to only include the shows they want.   

The next step is to enable all these podcasts to support LIT and then V4V.  Once they are LIT we can grab the date/start timings and create a dynamic schedule.   

### DRN1 Radio in Australia Publisher 'Radio' Feed on TrueFans. 

<img width="1295" alt="TrueFans___Publishers___DRN1" src="https://github.com/Podcastindex-org/podcast-namespace/assets/3492066/a4815224-9bda-415a-a60a-22329bbcb793">

The Publisher Feed below uses the medium=radio to help differentiate these podcasts from other types of audio.  It also helps apps know what type of podcast and can change the UI to reflect the content. In this case we simply changed episodes to shows and grouped them as a radio station.  We have more UI changes planned. 

### Example  - Radio Publisher Feed. 
```
<rss xmlns:podcast="https://podcastindex.org/namespace/1.0" xmlns:itunes="http://www.itunes.com/dtds/podcast-1.0.dtd" xmlns:content="http://purl.org/rss/1.0/modules/content/"xmlns:atom="http://www.w3.org/2005/Atom" version="2.0">
<channel>
<generator>TrueFans v1.1.0</generator>
<description/>
<lastBuildDate>Sat, 22 Jun 2024 06:56:48 GMT</lastBuildDate>
<itunes:image href="https://podfans.s3.eu-west-2.amazonaws.com/hosts-drn1-cover-art/83b843fb77f20ef7191f5f64.webp"/>
<title>DRN1</title>
<podcast:guid>667675b046a595e9f406360a</podcast:guid>
<podcast:medium>publisher</podcast:medium>
<podcast:remoteItem feedGuid="ac6d77c6-d735-56f2-925d-4e437914181f" feedUrl="https://feed.podtoo.com/distro/fQ6ViZ7y" medium="radio" title="Dykes On Bikes"/>
<podcast:remoteItem feedGuid="null" feedUrl="https://feed.podtoo.com/distro/3lSoRp-A" medium="radio" title="Out Loud Queer Music Radio"/>
<podcast:remoteItem feedGuid="397268a7-506f-50af-8272-3533bbe82b8d" feedUrl="https://feed.podtoo.com/distro/gHMSxsY0" medium="radio" title="The Netty Life"/>
<podcast:remoteItem feedGuid="6988a3de-ee40-51df-b7cf-8821964c57fd" feedUrl="https://feed.podtoo.com/distro/GG3awxpl" medium="radio" title="Women In Business"/>
</channel>
</rss>
```

So each podcaster would need to add a <podcast:publisher> tag with a link in their RSS feed.  This would point to the radio publisher.  Apps could then read the RSS feed and see the podcast:publisher tag.  They would then extract the publisher feed with multiple remote items to produce a publisher page similar to the one above.   

### Podcast with a Publisher Feed Link
```
<channel>
    <title><![CDATA[It's A Mood]]></title>
    <description>A value4value happenstance music show.</description>
    <link>https://example.org/itsamood</link>
    <generator>Sovereign Feeds</generator>
    <podcast:person href="https://example.org/mikeneumann" group="cast" role="host" img="https://example.org/mikeneumann/image.jpg">Mike Neumann</podcast:person>
    <podcast:guid>469b403f-db2d-574c-9db9-96dbb4f6561c</podcast:guid>
    <podcast:medium>podcast</podcast:medium>
    <podcast:publisher>
        <podcast:remoteItem medium="publisher" feedGuid="003af0a0-6a45-55cf-b765-68e3d349551a" feedUrl="https://agilesetmedia.com/assets/static/feeds/publisher.xml"/>            
    </podcast:publisher>
    <item>
        <title><![CDATA[Runnin']]></title>
        <pubDate>Wed, 03 Apr 2024 02:06:28 +0000</pubDate>
        ...
    </item>
</channel>
```

**Reply by nathangathright** [reply]
To address @yoshimo’s question, a radio could use the same live stream for all their shows. As long as each show has its own Live Item tag, it doesn’t make a difference if they’re all sharing the same stream. I believe jumping back to earlier parts of the live broadcast is feasible with HLS streams. Otherwise, listeners would have to wait until the broadcast is published as a normal podcast episode.

**Comment by agates** [comment]
I don't think it needs to be at the publisher level, personally.  Publishers are for aggregate feed ownership and mixing the two concepts is pretty tricky.

The suggested radio medium in another discussion can be used as a "24/7 station" and we could add a remoteItem to that liveItem to indicate "now playing" and link to the current feed being played.

We could also add a tag to the liveItem (`<podcast:nextUp>`?) that tells which feed/liveItem with a remoteItem to switch to next when the app receives a liveEnd podping.  This would be the equivalent of a "raid" on twitch where the streamer sends their viewers to another channel when they end their stream.

I think both of those options have merits for discovery.  One fulfills the radio discovery method and one fulfills the typical live stream scenario.  You can follow the radio feed and/or individual feeds.

**Reply by agates** [reply]
And if we just want a list of liveItems a playlist can fulfill that.

Where the radio medium comes in is it tells the app it's supposed to be always online and can be a composite of other live streams with links to them (and valueTimeSplit doesn't work because there's no absolute time to sync to).

Additionally, I think the app would like a hint to offer the user to disable notifications every time the 24/7 stream changes shows.  It's a pretty annoying user experience if not expected.

**Reply by nathangathright** [reply]
> I don't think it needs to be at the publisher level, personally. Publishers are for aggregate feed ownership and mixing the two concepts is pretty tricky.

That’s a good point. Conflating this idea with ownership wasn’t my intention. Apps could implement this idea for any playlist of feeds (whether that’s publishers, `medium="podcastL"` feeds, or a user-generated playlist on TrueFans) without any new tags. But a more explicit `podcast:nextUp` tag would be more straightforward.

**Comment by samsethi** [comment]
> The suggested radio medium in another discussion can be used as a "24/7 station" and we could add a remoteItem to that liveItem to indicate "now playing" and link to the current feed being played.

Love this idea. UI wise this is critical for radio stations to have it top of the website and very visible to users. 

> We could also add a tag to the liveItem (<podcast:nextUp>?) that tells which feed/liveItem with a remoteItem to switch to next when the app receives a liveEnd podping. This would be the equivalent of a "raid" on twitch where the streamer sends their viewers to another channel when they end their stream.

Love this too.
